olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp

```diff
@@ -61,8 +61,8 @@ DataResponse DataRepository::GetVersionedTile(
     const TileRequest& request, int64_t version,
     client::CancellationContext context,
     const client::OlpClientSettings& settings) {
-  auto response = PartitionsRepository::GetPartitionForVersionedTile(
-      catalog, layer_id, request, version, context, settings);
+  auto response = PartitionsRepository::GetAggregatedTile(
+      catalog, layer_id, context, request, version, settings);
```

Comment:
GetAggregatedTile can return ancestor instead of requested tile. You can check is found tile the same as in the request, but, I think, better to make similar to `PartitionsRepository::GetAggregatedTil` method which returns exactly what you want.